#!/usr/bin/env python3
PACKAGE = "val_soft_estop_monitor"

from dynamic_reconfigure.parameter_generator_catkin import *

# initialize parameter generator to define parameters
gen = ParameterGenerator()

# add parameters to generator
gen.add("debug",                      bool_t,   0, "Debug for printing",                                 False)
gen.add("monitor_velocity",           bool_t,   0, "Flag to monitor velocity",                           True)
gen.add("joint_velocity_delta_limit", double_t, 0, "Limit for velocity of any one joint",                5.0, 0.0, 5.0) # TODO default, min, max values?
gen.add("joint_velocity_timeout",	  double_t, 0, "Max threshold for time between two velocity points", 0.5, 0.0, 5.0) # TODO default, min, max values?
gen.add("monitor_torque",             bool_t,   0, "Flag to monitor torque",                             True)
gen.add("joint_torque_delta_limit",   double_t, 0, "Limit for torque applied by any one joint",          5.0, 0.0, 5.0) # TODO default, min, max values?
gen.add("joint_torque_timeout",       double_t, 0, "Max threshold for time between two torque points",   0.5, 0.0, 5.0) # TODO default, min, max values?

# generate necessary files
exit(gen.generate(PACKAGE, "val_soft_estop_monitor", "JointStateDeltaMonitorParams"))
